src/borg/archiver.py

```diff
@@ -1363,10 +1363,11 @@ def _list_repository(self, args, manifest, write):
         output_data = []
 
         for archive_info in manifest.archives.list_considering(args):
+            archive = Archive(repository, key, manifest, archive_info.name)
```

Comment:
You could also use the call keys concept from ItemFormatter to fix this; would avoid Archive vs. ArchiveInfo depending on branches.